src/Auth.js

```diff
@@ -128,22 +128,33 @@ Auth.prototype._loadRoles = function() {
     };
     // First get the role ids this user is directly a member of
     var query = new RestQuery(this.config, master(this.config), '_Role', restWhere, {});
-    return query.execute().then((response) => {
+    const p = Parse.Promise.when(
+      new Parse.Query(Parse.Role)
+        .equalTo('name', '_All_Role')
+        .first({ useMasterKey: true }),
+      query.execute()
+    );
+
+    // only using p for now so diff doesn't have a ton of whitespace changes.
+    return p.then((allRole, response) => {
       var results = response.results;
-      if (!results.length) {
-        this.userRoles = [];
-        this.fetchedRoles = true;
-        this.rolePromise = null;
-
-        cacheAdapter.role.put(this.user.id, this.userRoles);
-        return Promise.resolve(this.userRoles);
-      }
+      // if (!results.length) {
```

Comment:
every user is in the all group, so this optimization is no longer possible.  we have to see if the all group is a member of any other roles....